src/queue.c

```diff
@@ -6943,6 +6943,8 @@ _dispatch_root_queue_init_pthread_pool(dispatch_queue_global_t dq,
 		int pool_size, dispatch_priority_t pri)
 {
 	dispatch_pthread_root_queue_context_t pqc = dq->do_ctxt;
+	if (pqc == NULL)
```

Comment:
Ahhh hmmm. So I think the bug is actually here - init.c in the initialization of the root queues. 

do_ctxt = NULL for _DISPATCH_COOPERATIVE_ROOT_QUEUE_ENTRY is broken I think. It should be `_dispatch_root_queue_ctxt(_DISPATCH_ROOT_QUEUE_IDX(n, flags))`

Reply:
Assuming the root queue you are running into this for is of the new COOPERATIVE flavor. If you're running into this for other use cases, then there is another bug lying around.

Reply:
We might want to extend the Windows test suite to exercise cooperative flavored root queues as well.

Reply:
Yeah, I believe it was that root queue that was triggering this; I assumed that it was intentionally setting the context to null.  Adjusting the initializer seems fine, I should be able to change and test that.

Reply:
On darwin, `_dispatch_root_queue_ctxt` always evaluates to NULL since we don't use an internal workqueue. So this was an oversight on my part.